modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -262,6 +266,15 @@ private void removeLocalSubscriptionAndNotify(AndesSubscription subscription) th
         clusterNotificationAgent.notifySubscriptionsChange(subscription,
                 ClusterNotificationListener.SubscriptionChange.Closed);
 
+        // If there are no subscriptions for this queue, then delete it
+        if (storageQueue.getBoundSubscriptions().isEmpty() ) {
+
+            AndesContextInformationManager contextInformationManager = AndesContext.getInstance()
+                    .getAndesContextInformationManager();
+
+            contextInformationManager.deleteQueue(storageQueue);
```

Comment:
We should only delete storage queue for non-durable subscribers. Don't we need to check that here before deleting?